civihr_employee_portal/js/ta-documents-app.js

```diff
@@ -0,0 +1,50 @@
+angular.module('taDocuments', ['civitasks.appDocuments']).controller('myController', ['$scope', '$rootScope', '$rootElement', '$log', '$uibModal', 'DocumentService', 'FileService', 'config',
+  function($scope, $rootScope, $rootElement, $log, $modal, DocumentService, FileService, config) {
+    $scope.modalDocument = function(data, e) {
+      e && e.preventDefault();
+      var DocumentData = DocumentService.get({
+        'id': data.id
+      });
+
+      DocumentData.then(function(data) {
+        // Function to display the modal
+        openModalDocument(data[0]);
+      });
+    }
+
+    /**
+     * Opens Document Modal with or without document data
+     *
+     * @param {object} data
+     */
+    function openModalDocument(data) {
+      var modalInstance = $modal.open({
+        appendTo: $rootElement.find('div').eq(0),
+        templateUrl: config.path.TPL + 'modal/document.html?v=3',
```

Comment:
`config.path` is created as  follows:
```
  angular.module('civitasks.config',[]).constant('config', {
        ...
        path: {
            EXT: CRM.Tasksassignments.extensionPath,
            TPL: CRM.Tasksassignments.extensionPath + 'views/'
        },
```
And for views path we are using `templateUrl: config.path.TPL`.